src/analyses/apron/apronAnalysis.apron.ml

```diff
@@ -15,7 +15,25 @@ let spec_module: (module MCPSpec) Lazy.t =
       let name () = "apron"
     end
     in
-    (module Spec)
+    let narrowing_gas = GobConfig.get_int "ana.apron.narrowing_gas"
+    in
+    if (narrowing_gas > 0) then
+      let module Narrowed =
+      struct
+        module PolyhedraChainParams: Printable.ChainParams =
+        struct
+          let n () = narrowing_gas
+          let names = string_of_int
+        end
+        include NarrowingGas.DLifter (Spec) (PolyhedraChainParams)
+        module A = Spec.A
+        let access = Spec.access
+        let name = Spec.name
+      end
+      in
+      (module Narrowed)
+    else
+      (module Spec)
```

Comment:
This is in a slightly less ideal place than I imagined in #1494, but I can see why and it is fine.

Ideally, I'd like to just wrap this lifting around polyhedra, but not other Apron domains, somewhere in `ApronDomain`, and have the analysis be completely oblivious to the fact.
Because in some sense it's "just" a lifting of any `Lattice.S → Lattice.S`. But that's only so if things are fully properly abstract, but our `RelationDomain`/`ApronDomain` setup is far from that. So it'd require a large cleanup/refactoring to be able to do it in my dream way.

Reply:
Actually the `NarrowingGas.Dom` functor here (based on `WideningDelay`, etc) is essentially just that. It's the act of `unlift` and `lift` to do the operation that leads to the reset. And that's what `NarrowingGas.DLifter` used here exactly does.

It additionally needs all relational domain operations (instead of `Spec` operations) lifted, which needs more boilerplate. And it won't really gain anything from Goblint usage perspective.